pytorch_lightning/metrics/functional/sequence.py

```diff
@@ -0,0 +1,85 @@
+from collections import Counter
+
+import torch
+
+
+def _count_ngram(ngram_input_list: list, num_grams: int) -> Counter:
+    """Counting how many times each word appears in a given text with ngram
+
+    Args:
+        ngram_input_list: A list of translated text or reference texts
+        num_grams: gram value ranged 1 to 4
+
+    Return:
+        ngram_counter: a collections.Counter object of ngram
+    """
+
+    ngram_counter = Counter()
+
+    for i in range(1, num_grams + 1):
+        for j in range(len(ngram_input_list) - i + 1):
+            ngram_key = tuple(ngram_input_list[j : i + j])
+            ngram_counter[ngram_key] += 1
+
+    return ngram_counter
+
+
+def bleu_score(translate_corpus: list, reference_corpus: list, n: int = 4, smooth: bool = False) -> torch.Tensor:
+    """Calculate BLEU score of machine translated text with one or more references
+
+    Args:
+        translate_corpus: A list of lists of translated texts
+        reference_corpus: A list of lists of reference texts
+        n: Gram value ranged from 1 to 4 (Default 4)
+        smooth: Whether or not to apply smoothing – Lin et al. 2004
+
+    Return:
+        A Tensor with BLEU Score
+
+    Example:
+
+        >>> t = [["the", "FAST", "brown", "fox", "jumped", "over", "the", "lazy", "dog"]]
+        >>> r = [[["the", "quick", "brown", "fox", "jumped", "over", "the", "lazy", "dog"], ["the", "quick", "brown", "fox", "jumped", "over", "the", "the", "lazy", "cat"]]]
```

Comment:
```suggestion
        >>> t = ["the FAST brown fox jumped over the lazy dog".split(' ')]
        >>> r = [["the quick brown fox jumped over the lazy dog".split(' '), "the quick brown fox jumped over the the lazy cat".split(' ')]]
```
rather also use string sentence

Reply:
Okay, I totally missed that.